geode-core/src/test/java/org/apache/geode/internal/net/SocketCreatorJUnitTest.java

```diff
@@ -98,6 +106,65 @@ private void testBindExceptionMessageFormatting(InetAddress inetAddress) throws
     }
   }
 
+  @Test
+  public void configureSSLEngine() {
+    SSLConfig config = new SSLConfig.Builder().setCiphers("someCipher").setEnabled(true)
+        .setProtocols("someProtocol").setRequireAuth(true).setKeystore("someKeystore.jks")
+        .setAlias("someAlias").setTruststore("someTruststore.jks")
+        .setEndpointIdentificationEnabled(true).build();
+    SSLContext context = mock(SSLContext.class);
+    SSLParameters parameters = mock(SSLParameters.class);
+
+    SocketCreator socketCreator = new SocketCreator(config, context);
+
+    SSLEngine engine = mock(SSLEngine.class);
+    when(engine.getSSLParameters()).thenReturn(parameters);
+
+    final Object[] setProtocols = new Object[1];
+    doAnswer((Answer<Void>) invocation -> {
+      setProtocols[0] = invocation.getArgument(0);
+      return null;
+    }).when(engine).setEnabledProtocols(any(String[].class));
+
+    final Object[] setCiphers = new Object[1];
+    doAnswer((Answer<Void>) invocation -> {
+      setCiphers[0] = invocation.getArgument(0);
+      return null;
+    }).when(engine).setEnabledCipherSuites(any(String[].class));
+
+    socketCreator.configureSSLEngine(engine, "somehost", 12345, true);
+
+    verify(engine).setUseClientMode(isA(Boolean.class));
```

Comment:
Are you familiar with `ArgumentCaptors` in Mokito? I find it to be more readable approach to the argument validation than `doAnswer` approach. 
https://javadoc.io/doc/org.mockito/mockito-core/latest/org/mockito/Mockito.html#verify-T-

Reply:
I was not familiar with ArgumentCaptor.  Much simpler.